app/services/clients/ledger.client.test.js

```diff
@@ -0,0 +1,282 @@
+  type: 'payment',
+  amount: 100,
+  fee: 5,
+  net_amount: 95,
+  refund_summary_available: 100
+})
+
+const validTransactionSearchResponse = transactionDetailsFixtures.validTransactionSearchResponse({
+  gateway_account_id: '123456',
+  transactions: [
+    {
+      amount: 2000,
+      state: {
+        status: 'success',
+        finished: true
+      },
+      transaction_id: '222222',
+      created_date: '2018-09-22T10:14:15.067Z',
+      refund_summary_status: 'available',
+      refund_summary_available: 1850,
+      amount_submitted: 150,
+      type: 'payment',
+      card_brand: 'visa'
+    }
+  ]
+})
+
+const ledgerTransactionEventsFixture = {
+  events: [{
+    timestamp: 'some-iso-timestamp',
+    resource_type: 'PAYMENT',
+    data: {}
+  }]
+}
+
+const validTransactionSummaryResponse = transactionDetailsFixtures.validTransactionSummaryDetails({
+  account_id: '123456',
+  from_date: '2019-09-19T13:00:00.000Z',
+  to_date: '2019-09-22T00:00:00.000Z',
+  paymentCount: 2,
+  paymentTotal: 3500,
+  refundCount: 1,
+  refundTotal: 1000
+})
+
+class MockClient {
+  configure (baseUrl, options) {
+    configureSpy(baseUrl, options)
+  }
+
+  async get (url, description) {
+    let dataResponse
+
+    if (url.match(/\.*\/event/)) { dataResponse = ledgerTransactionEventsFixture }
+    else if (url.match(/\/transaction\//)) dataResponse = { validCreatedTransactionDetailsResponse }
+    else if (url.match(/\/report\/transactions-summary/)) { dataResponse = validTransactionSummaryResponse }
+    else if (url.match(/\/agreement/)) dataResponse = { validTransactionSummaryResponse }
+    else if (url.match(/\/payout/)) dataResponse = {}
+    else if (url.match(/\/transaction\?account_id.*limit_total/)) { dataResponse = validTransactionSearchResponse }
+    else dataResponse = {}
+
+    return Promise.resolve({ data: dataResponse })
+  }
+}
+
+function getLedgerClient () {
+  return proxyquire('./ledger.client', {
+    '@govuk-pay/pay-js-commons/lib/utils/axios-base-client/axios-base-client': { Client: MockClient }
+  })
+}
+
+describe('Ledger client', () => {
+  describe('transaction function', () => {
+    beforeEach(() => {
+      configureSpy.resetHistory()
+    })
+
+    it('should use default base URL when base URL has not been set', async () => {
+      const ledgerClient = getLedgerClient()
+
+      await ledgerClient.transaction('id', 'a-gateway-account-id', {})
+
+      expect(configureSpy.getCall(0).args[0]).to.equal('http://127.0.0.1:8006/v1/transaction/id?account_id=a-gateway-account-id')
+    })
+
+    it('should use configured base url', async () => {
+      const ledgerClient = getLedgerClient()
+
+      await ledgerClient.transaction('id', 'a-gateway-account-id', { baseUrl: 'https://example.com' })
+
+      expect(configureSpy.getCall(0).args[0]).to.equal('https://example.com/v1/transaction/id?account_id=a-gateway-account-id')
+    })
+  })
+
+  describe('transactionWithAccountOverride function', () => {
```

Comment:
In the code, this function seems to only use the default base URL option.
So not sure why this is working?
We can chat about this.

Reply:
Fixed by adding check to to see if options parameter contains baseUrl and using that if present.